src/IR/IR.hs

```diff
@@ -231,21 +234,26 @@ data Alt
   -- ^ @AltBinder v@ matches anything, and bound to name @v@.
   deriving (Eq, Show, Typeable, Data)
 
+newtype ExceptType
```

Comment:
Is there a reason you have `ExceptDefault` wrapping a literal here?

Reply:
I have this at the very beginning. I was thinking about later we might want to let user throw exception, so a literal can be used to distinct different type of Exception Error. That's why I make this for future implementation. But I remember you mentioned sslang users would not throw any exception. Maybe I can need to remove this completely?